pkg/controllers/vdb/altersubcluster_reconciler_test.go

```diff
@@ -0,0 +1,75 @@
+/*
+ (c) Copyright [2021-2024] Open Text.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ You may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package vdb
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	vapi "github.com/vertica/vertica-kubernetes/api/v1"
+	vmeta "github.com/vertica/vertica-kubernetes/pkg/meta"
+)
+
+var _ = Describe("altersubcluster_reconcile", func() {
+
+	It("should find subclusters to alter", func() {
+		vdb := vapi.MakeVDB()
+		annPri := map[string]string{
+			vmeta.ParentSubclusterTypeAnnotation: vapi.PrimarySubcluster,
+		}
+		annSec := map[string]string{
+			vmeta.ParentSubclusterTypeAnnotation: vapi.SecondarySubcluster,
+		}
+		const sbName = "sand"
+		vdb.Spec.Subclusters = []vapi.Subcluster{
+			{
+				Name: "sc1",
+				Type: vapi.SecondarySubcluster,
+			},
+			{
+				Name: "sc2",
+				Type: vapi.PrimarySubcluster,
+			},
+			{
```

Comment:
Add a comment here to say we don't support demote_subcluster now. The type of "sc3" should be demoted to a secondary subcluster in your test case.